tesseract_motion_planners/src/simple/profile/simple_planner_fixed_size_assign_plan_profile.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * @file simple_planner_default_plan_profile.cpp
+ * @brief
+ *
+ * @author Matthew Powelson
+ * @date July 23, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tesseract_motion_planners/simple/profile/simple_planner_fixed_size_assign_plan_profile.h>
+#include <tesseract_motion_planners/core/utils.h>
+
+namespace tesseract_planning
+{
+SimplePlannerFixedSizeAssignPlanProfile::SimplePlannerFixedSizeAssignPlanProfile(int freespace_steps, int linear_steps)
+  : freespace_steps_(freespace_steps), linear_steps_(linear_steps)
+{
+}
+
+CompositeInstruction SimplePlannerFixedSizeAssignPlanProfile::generate(const PlanInstruction& prev_instruction,
+                                                                       const PlanInstruction& base_instruction,
+                                                                       const PlannerRequest& request,
+                                                                       const ManipulatorInfo& global_manip_info) const
+{
+  InstructionInfo info1(prev_instruction, request, global_manip_info);
+  InstructionInfo info2(base_instruction, request, global_manip_info);
+
+  Eigen::MatrixXd states;
+  if (!info1.has_cartesian_waypoint && !info2.has_cartesian_waypoint)
+  {
+    const Eigen::VectorXd& jp = info2.extractJointPosition();
+    if (info2.instruction.getPlanType() == PlanInstructionType::LINEAR)
+      states = jp.replicate(1, linear_steps_ + 1);
```

Comment:
Is there a benefit to thinking about interpolation in terms of steps instead of number of states? My intuition would be to specify the number of states, so I always have to fight my confusion when I see `replicate(n_states + 1)`

Reply:
Not really, I have just gotten used to thinking in terms of steps because trajopt uses this terminology. Changeing things to use number of states versus steps would touch a lot of code which I would like to avoid at least for now. After the release we look at changing things.